DateTimeMate.go

```diff
@@ -30,31 +53,16 @@ func convertRelativeDateToActual(from string) string {
 // shrinkPeriod convert a period into a brief period
 // only allow one replacement per each period
 // Ex: 1 hour 2 minutes 3 seconds => 1h2m3s
-// FIXME: almost redundant code for plural & singular -- try to fix with strings.NewReplacer
 func shrinkPeriod(period string) string {
-	// plural
-	period = strings.Replace(period, "nanoseconds", "ns", 1)
-	period = strings.Replace(period, "microseconds", "us", 1)
-	period = strings.Replace(period, "milliseconds", "ms", 1)
-	period = strings.Replace(period, "seconds", "s", 1)
-	period = strings.Replace(period, "minutes", "m", 1)
-	period = strings.Replace(period, "hours", "h", 1)
-	period = strings.Replace(period, "days", "D", 1)
-	period = strings.Replace(period, "weeks", "W", 1)
-	period = strings.Replace(period, "months", "M", 1)
-	period = strings.Replace(period, "years", "Y", 1)
-
-	// singular
-	period = strings.Replace(period, "nanosecond", "ns", 1)
-	period = strings.Replace(period, "microsecond", "us", 1)
-	period = strings.Replace(period, "millisecond", "ms", 1)
-	period = strings.Replace(period, "second", "s", 1)
-	period = strings.Replace(period, "minute", "m", 1)
-	period = strings.Replace(period, "hour", "h", 1)
-	period = strings.Replace(period, "day", "D", 1)
-	period = strings.Replace(period, "week", "W", 1)
-	period = strings.Replace(period, "month", "M", 1)
-	period = strings.Replace(period, "year", "Y", 1)
+	for _, tuple := range abbrevMap {
+		period = strings.Replace(period, tuple[0], tuple[1], 1)
+	}
```

Comment:
I don't think this will work because the order returned by `range` is not guaranteed to be the same as the insertion order.

For example, if the order returned is from largest to smallest (eg: years to nanoseconds), then this could occur:

```shell
$ dtmate conv 4321s123456789ns hms.msusns -b
1h12m1s123millis456us788ns
```

Another example:

```shell
$ dtmate diff "2024-06-07T08:00:00Z" "2024-06-07T08:00:00.000123Z" -b
123micros
```

These types of errors occurred in development/testing on a different version of the code.  Therefore, I want to keep a replacement order of smallest time unit to largest which alleviates this condition/problem.

Reply:
Oh that's right, good catch indeed.

Maybe a comment above the [][]string could explain you need to respect order